### PR TITLE
Add Packer, Terraform configs to spin up an integrated Nomad, Consul, Vault cluster in AWS

### PR DESCRIPTION
**Pull request by rcgenova**
The goal is to enable a fully integrated sandbox environment on one or more cloud providers, starting with AWS. The Nomad clusters provisioned with this are Spark-enabled. A walk-through of the Nomad/Spark integration is also included. Nomad, Consul and Vault are co-located on the same nodes for efficiency and convenience. This is *not* meant to be a production-quality reference architecture.

### CODE REVIEW

File: terraform/shared/config/consul_upstart.conf [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+console log
+
+script
+  if [ -f "/etc/service/consul" ]; then
+    . /etc/service/consul
+  fi
+
+  # Make sure to use all our CPUs, because Consul can block a scheduler thread
+  export GOMAXPROCS=`nproc`
```
Comment (schmichael) [inline terraform/shared/config/consul_upstart.conf]: [nitpick] not needed; not a blocker though, just fyi

File: terraform/aws/Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+Vagrant.configure(2) do |config|
+
+  config.vm.box = "ubuntu/trusty64"
```
Comment (schmichael) [inline terraform/aws/Vagrantfile]: Just noticed the upstart script below, so 14.04 might be a good idea for now.

File: terraform/shared/config/nomad_client.hcl [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+bind_addr = "IP_ADDRESS"
+name = "nomad@IP_ADDRESS"
+
+# Enable the client
+client {
+  enabled = true
+  chroot_env {
+    "/bin"              = "/bin"
+    "/etc"              = "/etc"
+    "/home"             = "/home"
```
Comment (schmichael) [inline terraform/shared/config/nomad_client.hcl]: Do we need all of `/home`? If so this is fine, but we might want to explain why in a comment as I wouldn't recommend it for prod deploys. Would hate for it to get copied and pasted by people who don't need it.

File: terraform/aws/Vagrantfile (hunk 2 above)
Comment (rcgenova) [inline terraform/aws/Vagrantfile]: We should definitely upgrade to 16.04 and systemd in the future.

**Pull request merged**